src/lib.rs

```diff
@@ -1586,11 +1695,57 @@ mod test {
     #[test]
     #[cfg(feature = "std")]
     fn test_show() {
-        use std::string::ToString;
-        assert_eq!(format!("{}", _2), "2".to_string());
-        assert_eq!(format!("{}", _1_2), "1/2".to_string());
-        assert_eq!(format!("{}", _0), "0".to_string());
-        assert_eq!(format!("{}", Ratio::from_integer(-2)), "-2".to_string());
+        // Test:
+        // :b :o :x, :X, :?
+        // alternate or not (#)
+        // positive and negative
+        // padding
+        // does not test precision (i.e. truncation)
+        assert_eq!(&format!("{}", _2), "2");
+        assert_eq!(&format!("{}", _1_2), "1/2");
+        assert_eq!(&format!("{}", -_1_2), "-1/2"); // test negatives
+        assert_eq!(&format!("{}", _0), "0");
+        assert_eq!(&format!("{}", -_2), "-2");
+        assert_eq!(&format!("{:b}", _2), "10");
+        assert_eq!(&format!("{:b}", _1_2), "1/10");
+        assert_eq!(&format!("{:b}", _0), "0");
+        assert_eq!(&format!("{:#b}", _1_2), "0b1/0b10");
+        assert_eq!(&format!("{:010b}", _1_2), "0000001/10");
+        assert_eq!(&format!("{:#010b}", _1_2), "0b001/0b10");
+        let half_i8: Ratio<i8> = Ratio::new(1_i8, 2_i8);
+        assert_eq!(&format!("{:b}", -half_i8), "11111111/10");
+        assert_eq!(&format!("{:#b}", -half_i8), "0b11111111/0b10");
```

Comment:
Ew -- I find it unfortunate that negative numbers are formatted this way, especially since they'll overflow if you try to parse it back, but I know it's not your doing. If you're curious, it comes from core's [`int_base!`](https://github.com/rust-lang/rust/blob/f7af19c279b8b7ea3d2c21fcbd67164af8d5d968/src/libcore/fmt/num.rs#L130-L139) casting as unsigned.